### PR TITLE
build: verify package.json integrity

### PR DESCRIPTION
**Pull request by nedsalk**
Inspired by [this comment](https://github.com/FuelLabs/fuels-ts/pull/1374#discussion_r1376847563), I created a script which verifies the integrity of all `package.json`s of _non-private_ packages.

I also made the `@fuel-ts/testcases` package private as it's only used in `devDependencies`, except for the `fuels` package where it was set as a dependency but nothing was being exported from it so I removed it from its dependencies. It had some mismatches with the root `package.json`, but I didn't see the need to actually fix it to match `package.json` and opted to make it private. I suppose a refactor could make this package go away completely, but I didn't investigate much.

### CODE REVIEW

**Comment by Torres-ssf** [comment]
Is this CI check really necessary? This seems a little overkill to me

**Comment by nedsalk** [comment]
> @nedsalk Is this CI check really necessary? It seems a little overkill to me.

I see it as a win. Whoever updates node or pnpm in the future or creates a new package that differs in the `author` field will get a failing CI. It just automates @arboleya's comment and programmatically ensures a bit more consistency.

File: scripts/lint-package-jsons.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+const faultyPackageJsons = execSync('find packages/ apps/ -name package.json')
+  .toString()
+  .split('\n')
+  .filter(
+    (path) =>
+      path !== '' &&
+      !path.includes('/forc/') &&
+      !path.includes('/fuel-core/') &&
+      !path.match(/\/\..*\//) // internal folders like .next/package.json, .vitepress/package.json
+  )
```
Comment (arboleya) [inline scripts/lint-package-jsons.ts:34]: > [!WARNING]
> Nits-only. Feel free to ignore it.

```suggestion
const faultyPackageJsons = globSync(`{packages,apps}/*/package.json`)
  .filter((p) => !/\/forc|fuel-core\//.test(p))
```

---

If you fancy code/output aesthetics:

<details>
<summary>See code</summary>

```ts
import { error } from 'console';
import { readFileSync } from 'fs';
import { globSync } from 'glob';

interface PackageJson {
  private: boolean;
  author: string;
  engines: {
    node: string;
    pnpm: string;
  };
  packageManager: string;
}

const {
  author,
  engines: { node, pnpm },
  packageManager,
} = JSON.parse(readFileSync('package.json', 'utf-8')) as PackageJson;

const expectedAuthor = 'Fuel Labs <contact@fuel.sh> (https://fuel.network/)';

if (author !== expectedAuthor) {
  error(`The package.json's author field should be '${expectedAuthor}'`);
  process.exit(1);
}

const faultyPackageJsons = globSync(`{packages,apps}/*/package.json`)
  .filter((path) => !/\/forc|fuel-core\//.test(path))
  .filter((path) => {
    const json = JSON.parse(readFileSync(path, 'utf-8')) as PackageJson;
    return json.private
      ? false
      : json.author !== author ||
          json.engines.node !== node ||
          json.packageManager !== packageManager ||
          json.engines.pnpm !== pnpm;
  });

if (faultyPackageJsons.length) {
  const expectedConfigs = {
    author,
    engines: { node, pnpm },
    packageManager,
  };

  error(
    [
      `Expected configs:`,
      JSON.stringify(expectedConfigs, null, 2),
      `\nPlease review and correct these files:`,
      faultyPackageJsons.map((f) => ` - ${f}`).join('\n'),
    ].join('\n')
  );

  process.exit(1);
}
```
</details>

errors

**Review by arboleya — DISMISSED** [review]

File: scripts/lint-package-jsons.ts (hunk 1 above)
Comment (nedsalk) [reply scripts/lint-package-jsons.ts:34]: I fancy your suggestion!

**Review by arboleya — APPROVED** [review]

**Pull request merged**